### PR TITLE
docs: Improve clarity for Radix and complete AVL docs

### PR DESCRIPTION
**Pull request by 4ndrelim**

### CODE REVIEW

File: src/main/java/dataStructures/avlTree/AVLTree.java [hunk 1]
```diff
@@ -372,6 +372,10 @@ private T successor(Node<T> node) {
         return null;
     }
 
+
```
Comment (yeoshuheng) [inline src/main/java/dataStructures/avlTree/AVLTree.java:375]: Should we add this to all the current implementations (eg. segment portions of code based on it's purpose) for consistency?

File: src/main/java/algorithms/sorting/radixSort/RadixSort.java [hunk 2]
```diff
@@ -16,9 +16,9 @@ public class RadixSort {
      * @return The value of the digit in the number at the given segment.
      */
     private static int getSegmentMasked(int num, int segment) {
-        // Bit masking here to extract each segment from the integer.
-        int mask = ((1 << NUM_BITS) - 1) << (segment * NUM_BITS);
-        return (num & mask) >> (segment * NUM_BITS);
+        // bit masking here to extract each segment from the integer.
```
Comment (yeoshuheng) [inline src/main/java/algorithms/sorting/radixSort/RadixSort.java:19]: Actually I've been thinking - for students taking CS2040S, they have probably not touched CS2100 yet, so the concept of bit representation might be a bit complex for them. 

Would it be better to maybe insert more helpful links regarding how computers represent numbers (eg. float, integer).

Maybe we could add this link: https://cheever.domains.swarthmore.edu/Ref/BinaryMath/NumSys.html or somethign else to help clarify things.

**Review by yeoshuheng — COMMENTED** [review]
Have only reviewed the portion on Radix sort.
Have a few suggestions, but they don't need to be followed :)

LGTM.

**Review by yeoshuheng — COMMENTED** [review]
AVL Portion LGTM too.

File: src/main/java/algorithms/sorting/radixSort/RadixSort.java (hunk 2 above)
Comment (4ndrelim) [reply src/main/java/algorithms/sorting/radixSort/RadixSort.java:19]: added

**Review by kaitinghh — APPROVED** [review]
LGTM!

**Pull request merged**